test/core/memory_trap.wast

```diff
@@ -31,3 +31,93 @@
 (assert_trap (invoke "store" (i32.const 0x80000000) (i32.const 13)) "out of bounds memory access")
 (assert_trap (invoke "load" (i32.const 0x80000000)) "out of bounds memory access")
 (assert_return (invoke "grow_memory" (i32.const 0x10001)) (i32.const -1))
+
+
+(module
+  (memory 1)
+
+  (func (export "i32.store16") (param $a i32) (param $v i32)
+    (i32.store16 (get_local $a) (get_local $v))
+  )
+  (func (export "i32.store") (param $a i32) (param $v i32)
+    (i32.store (get_local $a) (get_local $v))
+  )
+  (func (export "i64.store16") (param $a i32) (param $v i64)
+    (i64.store16 (get_local $a) (get_local $v))
+  )
+  (func (export "i64.store32") (param $a i32) (param $v i64)
+    (i64.store32 (get_local $a) (get_local $v))
+  )
+  (func (export "i64.store") (param $a i32) (param $v i64)
+    (i64.store (get_local $a) (get_local $v))
+  )
```

Comment:
I suggest testing f32 and f64 stores too, since it's not uncommon for JIT implementations to handle those separately from integer stores.